### DISCUSSION TITLE
Multiple relationships with the same model

### DISCUSSION THREAD

**Discussion by auctumnus** [prisma/prisma#11392] [Q&A] [answered]
Hello! I have a schema with models like:
```prisma
model User {
    id       String @id @default(cuid())
    username String @unique
}

model Foo {
    id  String @id @default(cuid())
    bar String

    createdBy     User
    lastUpdatedBy User
}
```

This, of course, makes an ambiguous relation, the compiler prompting me to add a `@relation("name")`, but while it might make sense to have a `User.createdFoos`, it doesn't really make any sense to have, say, a `User.lastUpdatedFoosByThisUser` or something like that.

Is there a way to work around this, or a better way to model this information?

**Comment by austincrim** [comment, marked as answer]
Hey @auctumnus 👋 ,

I would most likely stick with the model you have already, just adding the tweaks that were suggested by the Prisma language tools.

```prisma
model User {
  id              String @id @default(cuid())
  username        String @unique
  createdFoos     Foo[]  @relation(name: "createdByUser")
  lastUpdatedFoos Foo[]  @relation(name: "lastUpdatedByUser")
}

model Foo {
  id  String @id @default(cuid())
  bar String

  createdBy     User   @relation(name: "createdByUser", map: "createdBy_fkey", fields: [userId], references: [id])
  lastUpdatedBy User   @relation(name: "lastUpdatedByUser", map: "lastUpdatedBy_fkey", fields: [userId], references: [id])
  userId        String
}
```

This makes the most sense to me but you can change any of the names to your liking. You can read more about `@relation` [in our docs](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#relation) if you are interested.

Let me know if you have any questions!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
